packages/admin/cms-admin/src/userPermissions/hooks/currentUser.tsx

```diff
@@ -0,0 +1,55 @@
+import { gql, useQuery } from "@apollo/client";
+import React from "react";
+
+import { GQLCurrentUserPermission } from "../../graphql.generated";
+import { GQLCurrentUserQuery } from "./currentUser.generated";
+
+const CurrentUserContext = React.createContext<CurrentUser | undefined>(undefined);
+
+export type UserPermission = string;
+
+export type ContentScope = {
+    [key: string]: string;
+};
+
+export class CurrentUser {
+    name = undefined;
+    email = undefined;
+    language = undefined;
+    permissions: GQLCurrentUserPermission[] = [];
+    contentScopes: ContentScope[] = [];
+    isAllowed(permission?: UserPermission): boolean {
+        if (this.email === undefined) return false;
+        if (!permission) return true;
+        return this.permissions.some((p) => p.permission === permission);
+    }
+}
+
+export const CurrentUserProvider: React.FC = ({ children }) => {
+    const { data, error } = useQuery<GQLCurrentUserQuery>(gql`
+        query CurrentUser {
+            currentUser {
+                name
+                email
+                contentScopes
+                permissions {
+                    permission
+                }
+            }
+            userPermissionsAvailablePermissions
+        }
+    `);
+
+    if (error) {
+        return <>Error while loading user: {error.message}</>;
+    }
+    const currentUser = new CurrentUser();
```

Comment:
This always creates a user, even when the user data is still loading. Is this a good idea?

Reply:
That way useCurrentUser always return a CurrentUser-Object instead of undefined which makes dealing with the userCurrentUser-hook a bit easier. Though the object can be empty it might spare you some if's, e.g. when looping through the contentScopes.
For me this implementation is okay and I don't see any disadvantages.

Reply:
The CurrentUserProvider now shows a loading banner, I think this is okay since the admin is more or less unusable without a user: 4ba452c9e147d8717de431c5230b9d1d76d93295